Updated App icon for macOS 26

ShadowPaint-SP opened the pull request:
Tested the Icon in macos 26 and macos 15.1 successfully.

This makes the Icon now compatible with macOS 26 Liquid Glass.

This fixes #4149

ShadowPaint-SP committed:
Added macOS 26 Liquid Glass Icon Support. See:  #4149

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP commented:
[PrismLauncher Exports.zip](https://github.com/user-attachments/files/22444032/PrismLauncher.Exports.zip)
For reference here are the icons exported ad png from the Icon Composer App

iblowmymind commented on cmake/MacOSXBundleInfo.plist.in:
```diff
@@ -21,7 +21,9 @@
     <string>${MACOSX_BUNDLE_INFO_STRING}</string>
     <key>CFBundleIconFile</key>
-    <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+    <string>${Launcher_Name}</string>
```
You should use `${Launcher_Name}.icns` for CFBundleIconFile here.

ShadowPaint-SP replied:
are u sure? I looked in some other apps and all of them dont mention the .icns anymore. when building with Xcode the icns file is also not mentioned and when moving the .app file to an older machine (15.1) the old icon is correctly shown.

iblowmymind replied:
On a couple applications I've checked I've seen it used with the extension, actually I think it seems to work either way though. Up to you.

ShadowPaint-SP replied:
I See ok lets see what the maintainers have to say. But Thanks for ur Info :>

iblowmymind replied:
No problem!

iblowmymind commented:
Additionally we might want to look at replacing the old icns here with the variant generated by Xcode, but that means the icon will look "glassy" on all macOS versions and I'm not sure the rest of the developers would be okay with that.

Also, I've modified the proportions a little again and I feel like this version looks a little better **(still needs to be checked)**, and here is my Assets.car, PrismLauncher.icon and PrismLauncher.icns:
[Icons.zip](https://github.com/user-attachments/files/22450042/Icons.zip)

ShadowPaint-SP commented:
I would say no, the icns file shouldn't be touched as this one is only used for older macOS versions, thus shouldn't have the glass look.
And I am not such a fan of the Translucency on the default icon (light mode) as it looks washed out that way and I think we should make as little changes as possible.

iblowmymind commented:
Alright, that's understandable.

quackerex commented:
Can you guys use icons from @hw2007. It looks much better imo
<img width="320" height="320" alt="image" src="https://github.com/user-attachments/assets/c7368ba2-55f1-4ade-8e8d-d57e781c46ce" />

https://github.com/orgs/PrismLauncher/discussions/3910

iblowmymind commented:
Ah yeah, hadn't seen that as it's in discussions; looks much better as it was actually implemented with effort (I just took the existing svg from the repository lol) so we should use that one instead.

hw2007 commented:
> Ah yeah, hadn't seen that as it's in discussions; looks much better as it was actually implemented with effort (I just took the existing svg from the repository lol) so we should use that one instead.

Sorry, wasn't sure if discussion was the right place to put it or not, since I don't have the knowledge to actually implement it in a pull request. Would be cool to see it used though!

iblowmymind commented:
Up to the maintainers, of course, but I do think it looks great. @ShadowPaint-SP Any chance you could get a build of this icon file instead?

ShadowPaint-SP committed:
updated Assets.car to the icon suggestion from hw2007

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP commented:
yes this one looks great. made a commit for using this one. But still looking for feedback from the maintainers. :)

DidiDidi129 commented:
Hey, are there any updates? Doesn't look like it. I'd really like this.

TheKodeToad commented:
Hmm, seems these CAR files are really hard to inspect if you're not on macOS so not really sure what to do

Mick235711 commented:
Well, if it is of any help, the [CAR file viewer](https://github.com/insidegui/AssetCatalogTinkerer) under macOS shows these images:
<img width="3456" height="756" alt="CleanShot 2025-12-09 at 23 07 33@2x" src="https://github.com/user-attachments/assets/45bd045e-a744-4c7a-871f-f53079a7eb59" />

Personally, I think it looks fine, and also gives the launcher dark mode icon compatibility.

EDIT: built a local version with this PR applied, can confirm it worked with light, dark, and clear under macOS 26.1 (arm64).
<img width="1472" height="680" alt="CleanShot 2025-12-09 at 23 32 33@2x" src="https://github.com/user-attachments/assets/71ec4f1e-68ad-400c-b9b7-a63d17ba4f83" />
<img width="1470" height="678" alt="CleanShot 2025-12-09 at 23 31 35@2x" src="https://github.com/user-attachments/assets/5c855924-a498-4142-8659-88a70fc5a629" />
<img width="1472" height="680" alt="CleanShot 2025-12-09 at 23 32 50@2x" src="https://github.com/user-attachments/assets/bda7f627-2a7a-4f63-b77a-db6b42128b14" />

DioEgizio reviewed (commented):
why do we even need an Assets.car file? can't we just update the icon we already use for macOS to be liquid glass?

edit: oh it's for the dark/clear icons ic. But I'm pretty sure there must be a better way to this than to just add an Assets.car file into the repo that can't even be opened outside macOS

ShadowPaint-SP commented:
so for the macOS 26 icons to be recognized correctly the Assets.car is kind of required but instead of adding it to the project it would be possible to create it from an .icon file (which everyone can red) in the compilation step. 
Would that be a better solution?
This would be done with actool

iblowmymind commented:
> why do we even need an Assets.car file? can't we just update the icon we already use for macOS to be liquid glass?
> 

The new icon format that Liquid Glass requires contains multiple layers and can only be stored in an Assets.car file when built. icns files don't allow storing this information, and the .icon format is just a bundle containing JSON/SVG files that's not really intended for production, the icon needs to be baked into the Assets file. That's how Apple designed it, can't really do much about it!

~~If needed, instead of trusting an Assets.car from a third party, one can create a blank (Hello world) project in Xcode, drop in the icon file as the icon, build the app and extract its Assets.car and it should work just fine.~~
It looks like someone has come up with a command line solution since I looked into it:
```sh
actool PrismLauncher.icon \
	--compile . \
	--app-icon PrismLauncher \
	--enable-on-demand-resources NO \
	--development-region en \
	--target-device mac \
	--platform macosx \
	--enable-icon-stack-fallback-generation=disabled \
	--include-all-app-icons \
	--minimum-deployment-target 10.14 \
	--output-partial-info-plist /dev/null
```
This should build an Assets.car to be included in the app bundle, could be used for Actions automation.

ShadowPaint-SP commented:
Yes I already started testing that out yesterday. I will have a new commit ready some time today.

Ryex commented:
yes. we would much prefer building the binary asset in ci than keeping it in tree.

ShadowPaint-SP committed:
replaced binary with in ci building of Icons

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP commented:
I now added the compilation of the assets and also noticed that the actool command also creates an .icns file, so it would be possible to remove the prismlauncher.icns file from the tree. For now I just changed it so the .icns file is taken from the actool output. The PrismLauncher.icon file is really just a directory with the .svg layers and an icon.json which contains the settings for the Liquid Glass effect and icon setup.

DioEgizio commented on CMakeLists.txt:
```diff
@@ -414,7 +414,39 @@ if(UNIX AND APPLE)
+        set(ASSETS_OUT_DIR "${CMAKE_BINARY_DIR}/program_info")
+        set(GENERATED_ASSETS_CAR "${ASSETS_OUT_DIR}/Assets.car")
+		set(GENERATED_LAUNCHER_ICNS "${ASSETS_OUT_DIR}/${Launcher_Name}.icns")
+        set(ICON_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/${Launcher_Branding_macos_ICON}")
```
```suggestion
        set(ICON_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/${Launcher_Branding_MAC_ICON}")
```

DioEgizio commented on CMakeLists.txt:
```diff
@@ -414,7 +414,39 @@ if(UNIX AND APPLE)
+                    --include-all-app-icons
+                    --enable-on-demand-resources NO
+                    --target-device mac
+                    --minimum-deployment-target 10.13
```
this should be based on CMAKE_OSX_DEPLOYMENT_TARGET

DioEgizio commented on CMakeLists.txt:
```diff
@@ -414,7 +414,39 @@ if(UNIX AND APPLE)
+                    --target-device mac
+                    --minimum-deployment-target 10.13
+                    --platform macosx
+					> /dev/null
```
why is this going to /dev/null ?

ShadowPaint-SP replied:
when actool runs it wants to create a plist file and apart from creating that file it also prints its contents into the console stream. I thought that looked weird but I see would be nice for debugging my bad :) 

The console output looks somewhat like this btw:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
        <key>com.apple.actool.compilation-results</key>
        <dict>
                <key>output-files</key>
                <array>
                        <string>.../build/program_info/Assets.car</string>
                        <string>.../build/program_info/PrismLauncher.icns</string>
                        <string>/dev/null</string>
                </array>
        </dict>
</dict>
</plist>
```

DioEgizio commented on program_info/CMakeLists.txt:
```diff
@@ -30,7 +30,7 @@ set(Launcher_Desktop "program_info/${Launcher_AppID}.desktop" PARENT_SCOPE)
 set(Launcher_MetaInfo "program_info/${Launcher_AppID}.metainfo.xml" PARENT_SCOPE)
 set(Launcher_SVG "program_info/${Launcher_SVGFileName}" PARENT_SCOPE)
-set(Launcher_Branding_ICNS "program_info/prismlauncher.icns" PARENT_SCOPE)
+set(Launcher_Branding_macos_ICON "program_info/PrismLauncher.icon" PARENT_SCOPE)
```
```suggestion
set(Launcher_Branding_MAC_ICON "program_info/PrismLauncher.icon" PARENT_SCOPE)
```

ShadowPaint-SP commented on CMakeLists.txt:
```diff
@@ -414,7 +414,39 @@ if(UNIX AND APPLE)
     endif()
 
     # Add the icon
-    install(FILES ${Launcher_Branding_ICNS} DESTINATION ${RESOURCES_DEST_DIR} RENAME ${Launcher_Name}.icns)
```
I also just thought of that but did I by removing this line removed the icon of the Linux builds? as the actool is an Xcode thing in the linux building it shouldn't be there thus is skipped, so it is indeed needed. am I right?

ShadowPaint-SP committed:
implemented recommended changes -reverted .icns file instellation

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP commented:
ok so it seems like the ci build isnt generating any files. Running the build locally does generate them though.

This is how it should look like:
> when actool runs it wants to create a plist file and apart from creating that file it also prints its contents into the console stream. I thought that looked weird but I see would be nice for debugging my bad :)
> 
> The console output looks somewhat like this btw:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
>         <key>com.apple.actool.compilation-results</key>
>         <dict>
>                 <key>output-files</key>
>                 <array>
>                         <string>.../build/program_info/Assets.car</string>
>                         <string>.../build/program_info/PrismLauncher.icns</string>
>                         <string>/dev/null</string>
>                 </array>
>         </dict>
> </dict>
> </plist>
> ```

and this is the output in the ci
```
[52/588] Compiling asset catalog (/Users/runner/work/PrismLauncher/PrismLauncher/program_info/PrismLauncher.icon)
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.actool.compilation-results</key>
	<dict>
		<key>output-files</key>
		<array>
			<string>/dev/null</string>
		</array>
	</dict>
</dict>
</plist>
```

DioEgizio commented:
> ok so it seems like the ci build isnt generating any files. Running the build locally does generate them though.
> 
> This is how it should look like:
> > when actool runs it wants to create a plist file and apart from creating that file it also prints its contents into the console stream. I thought that looked weird but I see would be nice for debugging my bad :)
> > 
> > The console output looks somewhat like this btw:
> > 
> > ```
> > <?xml version="1.0" encoding="UTF-8"?>
> > <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> > <plist version="1.0">
> > <dict>
> >         <key>com.apple.actool.compilation-results</key>
> >         <dict>
> >                 <key>output-files</key>
> >                 <array>
> >                         <string>.../build/program_info/Assets.car</string>
> >                         <string>.../build/program_info/PrismLauncher.icns</string>
> >                         <string>/dev/null</string>
> >                 </array>
> >         </dict>
> > </dict>
> > </plist>
> > ```
> 
> and this is the output in the ci
> ```
> [52/588] Compiling asset catalog (/Users/runner/work/PrismLauncher/PrismLauncher/program_info/PrismLauncher.icon)
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
> 	<key>com.apple.actool.compilation-results</key>
> 	<dict>
> 		<key>output-files</key>
> 		<array>
> 			<string>/dev/null</string>
> 		</array>
> 	</dict>
> </dict>
> </plist>
> ```

try updating the macOS runner to macos-26 maybe

ShadowPaint-SP committed:
bumped macos runner version to 26

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP committed:
added mono to brew because macos 26 runner doesnt include it

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP commented:
ok macOS build works now. For the Nix builds I saw that the nix.yml changed since I created the PR what should I do there? And also what came to my mind as the nix builds include older Mac runners can I catch that somehow?

DioEgizio commented:
I mean build just failing on older macs isn't that great either :/

ShadowPaint-SP committed:
removed unnecessary Option from actool command and failsafe attempt for older versions

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

Ryex commented on CMakeLists.txt:
```diff
@@ -438,7 +437,9 @@ if(UNIX AND APPLE)
         )
         add_custom_target(compile_assets ALL DEPENDS "${GENERATED_ASSETS_CAR}")
-        install(FILES "${GENERATED_ASSETS_CAR}" DESTINATION "${RESOURCES_DEST_DIR}")
+        if(EXISTS "${GENERATED_ASSETS_CAR}")
```
this test will only run at configure time when the file hasn't been generated yet.

ShadowPaint-SP committed:
xCode version check

Signed-off-by: Richard Voigtmann <richard.v.voigtmann@gmail.com>

ShadowPaint-SP commented:
ok can anyone help me identify why Test  7: ResourceFolderModel fails? I cant figure it out

DioEgizio commented:
> ok can anyone help me identify why Test  7: ResourceFolderModel fails? I cant figure it out

it's randomly failing sometimes recently, not related to your PR

DioEgizio commented on CMakeLists.txt:
```diff
@@ -416,6 +416,52 @@ if(UNIX AND APPLE)
+            add_custom_target(compile_assets ALL DEPENDS "${GENERATED_ASSETS_CAR}")
+            install(FILES "${GENERATED_ASSETS_CAR}" DESTINATION "${RESOURCES_DEST_DIR}")
+        else()
+            message(WARNING "Xcode ${XCODE_VERSION} is too old. Minimum required version is 26.0")
```
```suggestion
            message(WARNING "Xcode ${XCODE_VERSION} is too old. Minimum required version is 26.0. Not compiling liquid glass icons.")
```

ShadowPaint-SP committed:
Add suggestion from code review

Co-authored-by: DioEgizio <83089242+DioEgizio@users.noreply.github.com>
Signed-off-by: Richard Voigtmann <71901885+ShadowPaint-SP@users.noreply.github.com>

TheKodeToad commented on .github/actions/setup-dependencies/macos/action.yml:
```diff
@@ -14,7 +14,7 @@ runs:
       run: |
         brew update
-        brew install ninja extra-cmake-modules temurin@17
+        brew install ninja extra-cmake-modules temurin@17 mono
```
Confused why we need dotnet all of a sudden
Edit: oh I guess it could be related to updating macos... somehow

ShadowPaint-SP replied:
That was needed for the macOS ci. as of [macOS-26](https://github.com/actions/runner-images/blob/main/images/macos/macos-26-arm64-Readme.md) mono isn't pre installed and the build failed when it wanted to run a mono command so I added it. in older versions mono was there by default.

Ryex reviewed (approved):
the `if(EXISTS ` this test will only run at configure time when the file hasn't been generated yet.

TheKodeToad commented:
Considering it's approved can't we add the backport tag 🥺

Trial97 commented:
My only concern is that because we bump the macos runners to not break any older versions

ShadowPaint-SP commented:
Of course I am not that deep into it but in theory as the OSX Deployment target didn't change and the only difference to my first implementation is that the .car file is compiled on build it should still work on older versions. sadly I don't have my older Mac anymore so I cant try it out :(
